src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1035,7 +1035,9 @@ function _vcat(dfs::AbstractVector{<:AbstractDataFrame})
     length(header) == 0 && return DataFrame()
     cols = Vector{AbstractVector}(undef, length(header))
     for (i, name) in enumerate(header)
-        data = [df[name] for df in dfs]
+        # TODO: replace with commented out code after getindex deprecation
+        # data = [df[name] for df in dfs]
+        data = [df isa DataFrame ? df[name] : df[:, name] for df in dfs]
```

Comment:
Wouldn't it be better to always use `view` here (even if that's temporary)? Same for `completecases` and maybe other places.

Reply:
We could do this, but this would be breaking.
That is: currently `df[name]` when `df` is a `SubDataFrame` produces a copy and so I used `df[:, name]` to still make a copy temporarily (it will be a view after `getindex` deprecation).

Do you think making a `view` now would be better?

Reply:
But here it's an implementation detail which shouldn't have any user-visible consequences, right?

Reply:
Ah - right. I mixed it up with `hcat!` (this PR is really long). I will push a PR changing copy to `view` in places where it is not user visible.

Reply:
I have pushed a PR implementing what you have requested. Actually it is a good way to test if something will not break in the future. The only limitation we have is with `vcat` with which you have started (I have noted it in the code).